cmd/bootstrap/cmd/partner_infos.go

```diff
@@ -184,17 +182,39 @@ func executeGetNodeInfoScript(ctx context.Context, env templates.Environment, cl
 	return info, nil
 }
 
-// parseNodeInfo convert node info retrieved from
-func parseNodeInfo(info cadence.Value) (*NodePubInfo, error) {
+// parseNodeInfo convert node info retrieved from cadence script
+func parseNodeInfo(info cadence.Value) (*bootstrap.NodeInfoPub, error) {
 	fields := info.(cadence.Struct).Fields
-	role := flow.Role(fields[roleField].(cadence.UInt8))
-	return &NodePubInfo{
-		Role:          role.String(),
+	nodeID, err := flow.HexStringToIdentifier(string(fields[idField].(cadence.String)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert flow node ID from hex string to identifier (%s): %w", string(fields[idField].(cadence.String)), err)
+	}
+
+	b, err := hex.DecodeString(string(fields[networkingKeyField].(cadence.String)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode network public key hex (%s): %w", string(fields[networkingKeyField].(cadence.String)), err)
+	}
+	networkPubKey, err := crypto.DecodePublicKey(crypto.ECDSAP256, b)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode network public key: %w", err)
+	}
+
+	b, err = hex.DecodeString(string(fields[stakingKeyField].(cadence.String)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode staking public key hex (%s): %w", string(fields[stakingKeyField].(cadence.String)), err)
+	}
+	stakingPubKey, err := crypto.DecodePublicKey(crypto.BLSBLS12381, b)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode staking public key: %w", err)
+	}
+
+	return &bootstrap.NodeInfoPub{
+		Role:          flow.Role(fields[roleField].(cadence.UInt8)),
 		Address:       string(fields[networkingAddressField].(cadence.String)),
-		NodeID:        string(fields[idField].(cadence.String)),
-		Stake:         fields[tokensStakedField].(cadence.UFix64).String(),
-		NetworkPubKey: string(fields[networkingKeyField].(cadence.String)),
-		StakingPubKey: string(fields[stakingKeyField].(cadence.String)),
+		NodeID:        nodeID,
+		Stake:         uint64(fields[tokensStakedField].(cadence.UFix64)),
```

Comment:
⚠️ Type-casting `UFix64` to `uint64` changes the value, because the decimal part is encoded in the lower bits. 

Since we don't care about the on-chain registration of stake for this script, we can just set this to the default constant weight value.

Reply:
Good catch, https://github.com/onflow/flow-go/pull/1988/commits/cba109821575449a48fc29c352bf60649e5a3249